File: age_calculator.c
#include <stdio.h>
#include <time.h>
#include <string.h>

void get_age_message(int age, char *message) {
    if (age < 30) {
        strcpy(message, "You are young!");
    } else if (age >= 30 && age < 67) {
        strcpy(message, "You are middle-aged.");
    } else if (age >= 67 && age <= 80) {
        strcpy(message, "You are aging and should travel the world!");
    } else {
        strcpy(message, "You should relax and enjoy your life!");
    }
}

int main() {
    char name[100];
    int birth_year;
    char run_again[10];
    char message[100];

    do {
        time_t t = time(NULL);
        struct tm *tm_info = localtime(&t);
        int current_year = tm_info->tm_year + 1900;

        printf("\n========================================\n");
        printf("   Welcome to the Age Calculator!\n");
        printf("========================================\n\n");

        printf("Please enter your name: ");
        scanf("%s", name);

        printf("Please enter your year of birth: ");
        scanf("%d", &birth_year);

        int age = current_year - birth_year;

        get_age_message(age, message);

        printf("\n----------------------------------------\n");
        printf("Hello, %s!\n", name);
        printf("You are %d years old.\n", age);
        printf("%s\n", message);
        printf("----------------------------------------\n\n");
        
        printf("Would you like to run the program again? (yes/no): ");
        scanf("%s", run_again);

    } while (strcmp(run_again, "yes") == 0 || strcmp(run_again, "y") == 0);

    printf("\nThank you for using the Age Calculator. Goodbye!\n");

    return 0;
}

File: age_calculator.js
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

function getAgeMessage(age) {
    if (age < 30) {
        return "You are young!";
    } else if (age >= 30 && age < 67) {
        return "You are middle-aged.";
    } else if (age >= 67 && age <= 80) {
        return "You are aging and should travel the world!";
    } else {
        return "You should relax and enjoy your life!";
    }
}

function askQuestion(question) {
    return new Promise((resolve) => {
        rl.question(question, (answer) => {
            resolve(answer);
        });
    });
}

async function runCalculator() {
    const currentYear = new Date().getFullYear();

    console.log("\n========================================");
    console.log("   Welcome to the Age Calculator!");
    console.log("========================================\n");

    const name = await askQuestion("Please enter your name: ");
    const birthYear = parseInt(await askQuestion("Please enter your year of birth: "));

    const age = currentYear - birthYear;

    const message = getAgeMessage(age);

    console.log("\n----------------------------------------");
    console.log(`Hello, ${name}!`);
    console.log(`You are ${age} years old.`);
    console.log(message);
    console.log("----------------------------------------\n");

    const runAgain = await askQuestion("Would you like to run the program again? (yes/no): ");

    if (runAgain.toLowerCase() === "yes" || runAgain.toLowerCase() === "y") {
        await runCalculator();
    } else {
        console.log("\nThank you for using the Age Calculator. Goodbye!");
        rl.close();
    }
}

runCalculator();
